chains/ethereum/writer_methods.go

```diff
@@ -24,9 +24,10 @@ func (w *Writer) createErc20DepositProposal(m msg.Message) bool {
 	}
 
 	var data []byte
-	data = append(data, common.LeftPadBytes(m.Metadata[0].([]byte), 32)...)           // tokenId
 	data = append(data, common.LeftPadBytes(m.Metadata[1].([]byte), 32)...)           // recipient
 	data = append(data, common.LeftPadBytes(m.Metadata[2].(*big.Int).Bytes(), 32)...) // amount
+	data = append(data, common.LeftPadBytes(m.Metadata[0].([]byte), 32)...)           // tokenId
+	data = append(data, common.LeftPadBytes(big.NewInt(1).Bytes(), 32)...)            // tokenId length
```

Comment:
I think tokenID length should come before tokenID in data

Reply:
Honestly was a little confused. Please confirm that is whats happening here: https://github.com/ChainSafe/chainbridge-solidity/blob/22d99db56ceb58251fe9ca4be465ad845cd1904b/contracts/handlers/ERC20Handler.sol#L115-L118

Reply:
Can confirm, first instance of `tokenID := mload(0x40)` is loading the free memory pointer, sorry I should probably add more documentation